spec/uninitialized_spec.lua

```diff
@@ -0,0 +1,54 @@
+local driver = require 'pallene.driver'
+local util = require 'pallene.util'
+
+local function run_uninitialized(code)
+    assert(util.set_file_contents("test.pln", code))
+    local module, errs = driver.compile_internal("test.pln", "uninitialized")
+    return module, table.concat(errs, "\n")
+end
+
+local function assert_error(code, expected_err)
+    local module, errs = run_uninitialized(code)
+    assert.falsy(module)
+    assert.match(expected_err, errs, 1, true)
+end
+
+local missing_return =
+    "control reaches end of function with non-empty return type"
+
+
+describe("Uninitialized variable analysis: ", function()
+
+    teardown(function()
+        os.remove("test.pln")
+    end)
+
+    it("empty function", function()
+        assert_error([[
+            function fn(): integer
+            end
+        ]], missing_return)
+    end)
+
+    it("missing return in elseif", function()
+        assert_error([[
+            function getval(a:integer): integer
+                if a == 1 then
+                    return 10
+                elseif a == 2 then
+                else
+                    return 30
+                end
+            end
+        ]], missing_return)
+    end)
```

Comment:
I think we should a test case similar to the following:
```
function foo(cond: boolean): integer
  local x: integer
  while cond do
    x = 0
    cond = x == 0
  end
  return x
end
```

Reply:
Good idea